http/action/src/main/java/io/knotx/databridge/http/action/HttpAction.java

```diff
@@ -79,45 +83,60 @@
   private final String actionAlias;
   private final HttpActionOptions httpActionOptions;
   private final ResponsePredicatesProvider predicatesProvider;
+  private final ActionLogLevel logLevel;
   private static final ResponsePredicate IS_JSON_RESPONSE = ResponsePredicate
       .create(ResponsePredicate.JSON, result -> {
         throw new ReplyException(ReplyFailure.RECIPIENT_FAILURE, result.message());
       });
 
-  HttpAction(Vertx vertx, HttpActionOptions httpActionOptions, String actionAlias) {
+  HttpAction(Vertx vertx, HttpActionOptions httpActionOptions, String actionAlias,
+      ActionLogLevel actionLogLevel) {
     this.httpActionOptions = httpActionOptions;
     this.webClient = WebClient.create(io.vertx.reactivex.core.Vertx.newInstance(vertx),
         httpActionOptions.getWebClientOptions());
     this.endpointOptions = httpActionOptions.getEndpointOptions();
     this.actionAlias = actionAlias;
     predicatesProvider = new ResponsePredicatesProvider();
-    this.isJsonPredicate = this.httpActionOptions.getResponseOptions().getPredicates().contains(JSON);
+    this.isJsonPredicate = this.httpActionOptions.getResponseOptions().getPredicates()
+        .contains(JSON);
     this.isForceJson = httpActionOptions.getResponseOptions().isForceJson();
+    this.logLevel = actionLogLevel;
   }
 
   @Override
   public void apply(FragmentContext fragmentContext,
       Handler<AsyncResult<FragmentResult>> resultHandler) {
-    Single<FragmentResult> result = process(fragmentContext);
+    final ActionLogger actionLogger = ActionLogger.create(actionAlias, logLevel);
+    Single<FragmentResult> result = process(fragmentContext, actionLogger);
     result.subscribe(onSuccess -> {
       Future<FragmentResult> resultFuture = Future.succeededFuture(onSuccess);
       resultFuture.setHandler(resultHandler);
     }, onError -> {
-      Future<FragmentResult> resultFuture = Future.failedFuture(onError);
+      final Future<FragmentResult> resultFuture;
+      if (onError instanceof DecodeException) {
```

Comment:
I think that the logic below can be used for all exceptions. It is much better to get `_error` transition with some node log details.

Reply:
fixed